pkg/cli/config/adminkubeconfig/new_admin_kubeconfig.go

```diff
@@ -0,0 +1,263 @@
+package adminkubeconfig
+
+import (
+	"bytes"
+	"context"
+	"crypto/x509"
+	"fmt"
+	"reflect"
+	"time"
+
+	"github.com/openshift/library-go/pkg/crypto"
+	"github.com/openshift/library-go/pkg/operator/certrotation"
+	"github.com/openshift/oc/pkg/cli/config/refreshcabundle"
+	"github.com/spf13/cobra"
+	corev1 "k8s.io/api/core/v1"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apiserver/pkg/authentication/user"
+	"k8s.io/cli-runtime/pkg/genericclioptions"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/tools/clientcmd"
+	clientcmdapi "k8s.io/client-go/tools/clientcmd/api"
+	"k8s.io/client-go/util/cert"
+	cmdutil "k8s.io/kubectl/pkg/cmd/util"
+	"k8s.io/kubectl/pkg/util/i18n"
+	"k8s.io/kubectl/pkg/util/templates"
+)
+
+type NewAdminKubeConfigOptions struct {
+	RESTClientGetter genericclioptions.RESTClientGetter
+
+	genericclioptions.IOStreams
+}
+
+var (
+	newAdminKubeconfigLong = templates.LongDesc(i18n.T(`
+		Generate, make the server trust, and display a new admin.kubeconfig.
+
+		The key is produced locally and never hits disk.  The public half is pushed to the cluster
+		for the kube-apiserver to trust the locally created admin.kubeconfig.`))
+
+	newAdminKubeconfigExample = templates.Examples(`
+		# Generate a new admin kubeconfig
+		oc config new-admin-kubeconfig`)
+)
+
+func NewAdminKubeconfigOptions(restClientGetter genericclioptions.RESTClientGetter, streams genericclioptions.IOStreams) *NewAdminKubeConfigOptions {
+	return &NewAdminKubeConfigOptions{
+		RESTClientGetter: restClientGetter,
+		IOStreams:        streams,
+	}
+}
+
+func NewCmdConfigRefreshCABundle(restClientGetter genericclioptions.RESTClientGetter, streams genericclioptions.IOStreams) *cobra.Command {
+	o := NewAdminKubeconfigOptions(restClientGetter, streams)
+
+	cmd := &cobra.Command{
+		Use:                   "new-admin-kubeconfig",
+		DisableFlagsInUseLine: true,
+		Short:                 i18n.T("Generate, make the server trust, and display a new admin.kubeconfig."),
+		Long:                  newAdminKubeconfigLong,
+		Example:               newAdminKubeconfigExample,
+		Run: func(cmd *cobra.Command, args []string) {
+			cmdutil.CheckErr(o.Complete())
+			cmdutil.CheckErr(o.Validate(args))
+			r, err := o.ToRuntime()
+			cmdutil.CheckErr(err)
+			cmdutil.CheckErr(r.Run(context.TODO()))
+
+		},
+	}
+
+	o.AddFlags(cmd)
+
+	return cmd
+}
+
+func (o *NewAdminKubeConfigOptions) AddFlags(cmd *cobra.Command) {
+}
+
+func (o *NewAdminKubeConfigOptions) Complete() error {
+
+	return nil
+}
+
+func (o NewAdminKubeConfigOptions) Validate(args []string) error {
+	if len(args) > 0 {
+		fmt.Errorf("no arguments allowed")
```

Comment:
This is just https://pkg.go.dev/github.com/spf13/cobra#ExactArgs with zero.

Reply:
```go
	if err := cobra.ExactArgs(0)(cmd, args); err != nil {
		return err
	}
```

Doesn't seem to read easier to future maintainers to me.